_posts/2024-09-23-wai-aria.md

```diff
@@ -0,0 +1,81 @@
+
+ARIA is a set of rules and attributes that allow developers to define more refined semantics for different UI elements. Often, when JavaScript is added to user interactions, extra guidance is needed for screen readers and other accessibility tools. ARIA enables the use of non-semantic HTML, though this doesn't mean developers should aim to write non-semantic HTML. ARIA rules and attributes help accessibility tools convey to users what is happening on the page and what interactions are possible.
+
+## HTML vs. ARIA: Which Should You Use?
+
+When comparing HTML to ARIA, ARIA should always be a last resort. **If there is a semantic HTML element available, use that.** There’s no need for HTML like this: `<div role="navigation">...</div>`. Instead, you should use a native HTML element like `<nav>...</nav>`.
+
+Before using ARIA, check [the native HTML elements](https://developer.mozilla.org/en-US/docs/Web/HTML) to ensure none apply to your situation. A lot can be achieved with HTML alone.
+
+## How to Use ARIA
+
+As mentioned, ARIA consists of a set of roles and attributes. Let's go over what they are used for.
+
+### Roles
+
+ARIA roles make it possible to describe an element's function in a way that HTML can't. Some HTML elements have default roles. For example, `<input type="radio">` automatically has the role of "radio." Other elements have no role unless one is specifically assigned.
+
+ARIA roles include `tooltip`, `tab`, and `dialog`. They can be divided into six categories: (1) document structure roles, (2) widget roles, (3) landmark roles, (4) live region roles, (5) window roles, and (6) abstract roles. You can find the full list of roles on [MDN's page](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles).
+
+Many roles have equivalents in HTML, so there's generally no need to use `role="article"` when you can use the built-in `<article>` element. However, sometimes assigning a role is necessary. For instance, live region roles can signal to screen readers that certain elements require the user’s immediate attention.
+
+That said, it’s crucial to use `role="alert"` only when the user's immediate attention is genuinely required. Misusing it can disrupt the user experience by diverting the screen reader's focus. Just because you, as a developer, think something is urgent doesn’t mean the end user will feel the same way.
+
+### Attributes
+
+ARIA attributes modify an HTML element's state or properties, but these changes apply only to the accessibility tree, not the DOM. For instance, adding `aria-hidden="true"` or `aria-expanded="true"` won’t affect the visual page but will update the accessibility tree. This can hide elements from assistive technologies or indicate that an element controls another element that is currently expanded.
```

Comment:
Saying it doesn't apply to the DOM is not technically right as the properties and attributes are part of the DOM. I understand what you mean here but it should be written somehow differently.

The ARIA attributes by default don't affect the visual representation of the page unless the page includes CSS that applies styles based on ARIA attributes. E.g. `[aria-hidden="true"] { display: none }`